genson/src/main/java/com/owlike/genson/convert/RuntimeTypeConverter.java

```diff
@@ -58,4 +69,32 @@ private boolean isContainer(T obj) {
                 obj instanceof Collection ||
                 obj instanceof Map;
     }
+
+    private boolean isSimpleType(T obj) {
+      return obj instanceof Boolean ||
+          obj instanceof Number ||
+          obj instanceof String;
+    }
+
+    private void ensureNoCircularRefs(T obj, Context ctx) {
+      if (!isContainer(obj) && !isSimpleType(obj))
+      {
+        Set<Object> seen = ctx.get(CYCLE_KEY, Set.class);
```

Comment:
Should we use `IdentityHashMap` here? It is probably more efficient than a `HashSet`, and is pretty much designed for this particular purpose.

Reply:
Good point.